org.eclipse.dartboard/src/org/eclipse/dartboard/launch/LaunchShortcut.java

```diff
@@ -38,57 +44,80 @@
  * @author Jonas Hungershausen
  *
  */
+@SuppressWarnings("restriction")
 public class LaunchShortcut implements ILaunchShortcut {
 
 	@Override
 	public void launch(ISelection selection, String mode) {
+		// Get the active project from the
 		IProject selected = null;
 		if (selection instanceof StructuredSelection) {
 			Object firstElement = ((StructuredSelection) selection).getFirstElement();
-			if (firstElement instanceof IProject) {
-				selected = (IProject) firstElement;
+			if (firstElement instanceof IResource) {
+				selected = ((IResource) firstElement).getProject();
 			}
 		}
 
+		// Shouldn't happen
+		if (selected == null) {
+			MessageDialog.openError(null, Messages.Launch_ConfigurationRequired_Title,
+					Messages.Launch_ConfigurationRequired_Body);
+			return;
+		}
+
+		launchProject(selected, mode);
+	}
+
+	@Override
+	public void launch(IEditorPart editor, String mode) {
+		IResource resource = editor.getEditorInput().getAdapter(IResource.class);
+
+		if (resource == null) {
```

Comment:
launchProject(IProject project, String mode)  should handle this dialog

Reply:
As mentioned above this is incase the resource is null, and not if the project is null. As we are accessing getProject() of IResource the dialog should also be shown here if resource is null.

Reply:
IProject project = null;
		IEditorInput editorInput = editor.getEditorInput();
		if (editorInput instanceof IFileEditorInput) {
			project = ((IFileEditorInput) editorInput).getFile().getProject();
		}
		launchProject(project, mode);

Generally, file and project won't be null for  IFileEditorInput. You can do  null check for file inside the block

Reply:
Done.